src/rustup-cli/rustup_mode.rs

```diff
@@ -950,17 +945,34 @@ fn override_remove(cfg: &Cfg, m: &ArgMatches) -> Result<()> {
     Ok(())
 }
 
+const DOCS_DATA: &[(&'static str, &'static str, &'static str,)] = &[
+    // flags can be used to open specific documents, e.g. `rustup doc --nomicon`
+    // tuple elements: document name used as flag, help message, document index path
+    ("alloc", "The Rust core allocation and collections library", "alloc/index.html"),
+    ("book", "The Rust Programming Language book", "book/index.html"),
+    ("cargo", "The Cargo Book", "cargo/index.html"),
+    ("core", "The Rust Core Library", "core/index.html"),
+    ("edition-guide", "The Rust Edition Guide", "edition-guide/index.html"),
+    ("nomicon", "The Dark Arts of Advanced and Unsafe Rust Programming", "nomicon/index.html"),
+    ("proc_macro", "A support library for macro authors when defining new macros", "proc_macro/index.html"),
+    ("reference", "The Rust Reference", "reference/index.html"),
+    ("rust-by-example", "A collection of runnable examples that illustrate various Rust concepts and standard libraries", "rust-by-example/index.html"),
+    ("rustc", "The compiler for the Rust programming language", "rustc/index.html"),
+    ("rustdoc", "Generate documentation for Rust projects", "rustdoc/index.html"),
+    ("std", "Standard library API documentation", "std/index.html"),
+    ("test", "Support code for rustc's built in unit-test and micro-benchmarking framework", "test/index.html"),
+    ("unstable-book", "The Unstable Book", "unstable-book/index.html"),
+];
+
```

Comment:
Rather than hard-coding this list, is there a way we can load it from the toolchain?  I accept that might be hard currently since we'd have to parse a partial commandline loosely to ensure we knew which toolchain to use, and then go again once we'd loaded the docs list, but it'd mean that if new docs were added, we'd automatically pick them up.

Reply:
Yeah, I have also considered auto-generating vs hard-coding. Actually, the flags are got by running `cd ~/.rustup/toolchains/nightly-2018-12-31-x86_64-apple-darwin/share/doc/rust/html && ls */index.html | xargs -n1 echo | cut -d'/' -f1` :)

In addiction to the toolchain choosing problem, I think these should also be considered:

1. `rust doc` flags did not always have to match up with the directory name where the docs resides, that's why the document index path isn't auto-generated by appending `/index.html` to flags. This is also how things are handled at the moment.
2. Help messages cannot be easily auto-generated due different layouts of these docs
3. I think it may not be the case that every doc needs a `cargo doc` shortcut, in addition this list is not expected to change very often. Handpicking might be a better choice

Reply:
That sounds fair, thanks for explaining it.